ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/NettyApplicationEngine.kt

```diff
@@ -125,7 +125,11 @@ public class NettyApplicationEngine(
         environment.connectors.map { connector ->
             ServerBootstrap().apply {
                 configuration.configureBootstrap(this)
-                group(connectionEventGroup, workerEventGroup)
+
+                if (config().group() == null && config().childGroup() == null) {
+                    group(connectionEventGroup, workerEventGroup)
+                }
+
                 channel(connectionEventGroup.channel.java)
```

Comment:
If we already have an event group configured, we overwrite a channel class here that may be wrong because different event groups may have different channel classes. Not sure if it actually dangerous but there is a chance of incompatible channels produced from here.